executing/executing.py

```diff
@@ -509,9 +530,556 @@ def visit_ClassDef(self, node):
+                node = self.find_node(
+                lasti,
+                match_positions=("end_col_offset", "end_lineno"),
+                typ=ast.Attribute,
+            )
+            elif self.opname(lasti) == "CALL":
+                # A CALL instruction can be a method call, in which case the lineno and col_offset gets changed by the compiler.
+                # Therefore we ignoring here this attributes and searchnig for a Call-node only by end_col_offset and end_lineno. 
+                # This is save, because there can only be one method ending at a specific point in the source code.
+                # One closing ) only belongs to one method.
+                node = self.find_node(
+                    lasti,
+                    match_positions=("end_col_offset", "end_lineno"),
+                    typ=ast.Call,
+                )
+            else:
+                raise
+
+        # report KnownIssues
+
+        instruction = self.instruction(lasti)
+
+        if instruction.opname in ("COMPARE_OP", "IS_OP", "CONTAINS_OP") and isinstance(
+            node, self.types_cmp_issue
+        ):
+            if isinstance(node, self.types_cmp_issue_fix):
+                # this is a workaround for https://github.com/python/cpython/issues/95921
+                # we can fix cases with only on comparison inside the test condition
+                #
+                # we can not fix cases like:
+                # if a<b<c and d<e<f: pass
+                # if (a<b<c)!=d!=e: pass
+                # because we don't know which comparison caused the problem
+
+
+                comparisons = [
+                    n
+                    for n in ast.walk(node.test)
+                    if isinstance(n, ast.Compare) and len(n.ops) > 1
+                ]
+
+                assert_(comparisons, "expected at least one comparison")
+
+                if len(comparisons) == 1:
+                    node = comparisons[0]
+                else:
+                    raise KnownIssue(
+                        "multiple chain comparison inside %s can not be fixed" % (node)
+                    )
+
+            else:
+                # Comprehension and generators get not fixed for now.
+                raise KnownIssue("chain comparison inside %s can not be fixed" % (node))
+
+        if isinstance(node, ast.Assert):
+            # pytest assigns the position of the assertion to all expressions of the rewritten assertion.
+            # All the rewritten expressions get mapped to ast.Assert, which is the wrong ast-node.
+            # We don't report this wrong result.
+            raise KnownIssue("assert")
+
+        if any(isinstance(n, ast.pattern) for n in node_and_parents(node)):
+            # TODO: investigate
+            raise KnownIssue("pattern matching ranges seems to be wrong")
+
+        if instruction.opname == "STORE_NAME" and instruction.argval == "__classcell__":
```

Comment:
If I remember correctly the problem looks like this:
```
class X:
    # ... something
    if some_condition:
        def method(self):
            pass
```
this creates a `STORE_NAME` for the `__classcell__` and is mapped to the `if` statement. I didn't liked the Idea to put some kind of `STORE_NAME` with `__classcell__` can be mapped to *everything* to the verifier, because the last statement in the class could be (almost) everything.

But I also think now that it might not be such a bad idea. I am not sure about this.

Reply:
OK, I've reproduced with this script:

```python
import inspect
import dis

class X:
    if dis.get_instructions:
        def foo(self):
            return super()

code = inspect.currentframe().f_code
print(code.co_consts)  # To find the code for 'X'
for i in dis.get_instructions(code.co_consts[2], show_caches=True):
    print(i)
```

And towards the end there's:

```
Instruction(opname='STORE_NAME', opcode=90, arg=6, argval='__classcell__', argrepr='__classcell__', offset=48, starts_line=None, is_jump_target=False, positions=Positions(lineno=6, end_lineno=8, col_offset=4, end_col_offset=26))
```

and that Positions corresponds to the `if` block. Looks weird, we can leave it, maybe just expand the comment a bit more to explain.